ibis/pandas/core.py

```diff
@@ -403,15 +431,15 @@ def main_execute(
 
     # TODO: make expresions hashable so that we can get rid of these .op()
     # calls everywhere
-    params = {k.op() if hasattr(k, 'op') else k: v for k, v in params.items()}
+    for k, v in params.items():
```

Comment:
I think this is a little better and matches the original code
```
params = {Scope.make_scope(k.op if hasattr(k ,'op') else k, v, timecontext) for k,v in params.items}

scope.merge_scopes(params)
```

Reply:
Or even this:
```
params = {Scope.make_scope(getattr(k, 'op', k), v, timecontext) for k,v in params.items}

scope.merge_scopes(params)
```

Reply:
With merge_scopes added we could do this, sure.